src/binary/Builtin.cc

```diff
@@ -632,6 +632,45 @@ extern "C" {
     return INT2FIX(strcoll(RSTRING_PTR(str1), RSTRING_PTR(str2)));
   }
 
+  static int
+  tm_hash_value(VALUE hash, const char *key)
+  {
+    return NUM2INT(rb_hash_aref(hash, ID2SYM(rb_intern(key))));
+  }
+  
+  static struct tm *
+  hash_to_tm(VALUE hash)
+  {
+    static struct tm res;
+
+    res.tm_sec   = tm_hash_value(hash, "tm_sec");
+    res.tm_min   = tm_hash_value(hash, "tm_min");
+    res.tm_hour  = tm_hash_value(hash, "tm_hour");
+    res.tm_mday  = tm_hash_value(hash, "tm_mday");
+    res.tm_mon   = tm_hash_value(hash, "tm_mon");
+    res.tm_year  = tm_hash_value(hash, "tm_year");
+    res.tm_wday  = tm_hash_value(hash, "tm_wday");
+    res.tm_yday  = tm_hash_value(hash, "tm_yday");
+    res.tm_isdst = tm_hash_value(hash, "tm_isdst");
+
+    return &res;
+  }
+
+  static VALUE
+  strftime_wrapper(VALUE self, VALUE time, VALUE format)
+  {
+    char res[100];
+    struct tm *timeinfo;
+
+    Check_Type(format, T_STRING);
+    timeinfo = hash_to_tm(time);
+
+    if (strftime(res, sizeof(res), RSTRING_PTR(format), timeinfo)) {
+      return yrb_utf8_str_new(string(res));
+    } else {
+      return Qnil; // TODO: learn how to raise an exception
```

Comment:
exception is raised e.g. this way: https://github.com/yast/yast-ruby-bindings/blob/7f6545d5185aa3d3d49d907b5df08c3859d9c83b/src/binary/Yast.cc#L120 or https://github.com/yast/yast-ruby-bindings/blob/7f6545d5185aa3d3d49d907b5df08c3859d9c83b/src/binary/Yast.cc#L88